File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s=="":
            return 0
        max_sub_len=1
        l=[]
        for i in s:
            if i in l:
                list_len=len(l)
                ind=l.index(i)
                if max_sub_len<list_len:
                    max_sub_len=list_len
                if ind+1==list_len:
                    l=[i]
                else:    
                    l=l[ind+1:]+[i]
            else:
                l.append(i)              
        return max(max_sub_len,len(l))        
                    
                    

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        if(n==0) return 0;
        int maxCount=1;
        int count=1;
        map<char,int>map;
        map.insert({s[0],0});
        int i=1;
        int start=0;
        while(i<n){
           if(map.find(s[i])==map.end()){
               map[s[i]]=i;
               count++;
           }
            else if(map[s[i]]<start){
                map[s[i]]=i;
                    count++;
            }
           else{
               start=map[s[i]]+1;
               count=i-map[s[i]];
               map[s[i]]=i;
           } 
             maxCount=max(maxCount,count);
            i++;
        }
        return maxCount;
    }
};